src/lib.rs

```diff
@@ -11,6 +11,678 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+    sync::Arc,
+};
+
+use async_std::{sync::RwLock, task::block_on};
+use log::info;
+use mqtt::AsyncClient;
+use paho_mqtt::{self as mqtt, MQTT_VERSION_5};
+use protobuf::Enum;
+use up_rust::{ComparableListener, UAttributes, UCode, UMessage, UPriority, UStatus, UUri, UUID};
+
+pub mod rpc;
 pub mod transport;
 
-pub struct UPClientMqtt {}
+/// Constants defining the protobuf field numbers for UAttributes.
+// TODO: Convert this to a dynamically generated list.
+pub const ID_NUM: &str = "1";
+pub const TYPE_NUM: &str = "2";
+pub const SOURCE_NUM: &str = "3";
+pub const SINK_NUM: &str = "4";
+pub const PRIORITY_NUM: &str = "5";
+pub const TTL_NUM: &str = "6";
+pub const PERM_LEVEL_NUM: &str = "7";
+pub const COMMSTATUS_NUM: &str = "8";
+pub const REQID_NUM: &str = "9";
+pub const TOKEN_NUM: &str = "10";
+pub const TRACEPARENT_NUM: &str = "11";
+pub const PAYLOAD_NUM: &str = "12";
+
+// URI Wildcard consts
+const WILDCARD_AUTHORITY: &str = "*";
```

Comment:
I spoke about this with @evshary as well. These should probably go into up-rust.

Reply:
Yes I would agree, this will be stripped out once it gets into up-rust. but needed it here for completeness. Will add a todo